### PR TITLE
Allow Non-Blocking Socket Calls

### LINKED ISSUE #9: socket.accept() is blocking by default
After adding the Start/Poll method I setup some tests to see if other code could run even if there were no requests coming in.  Unfortunately, that wasn't the case.  Digging into it deeper I discovered that `socket.accept()` method is blocking by default.

Modifying the example code with a simple print will show the issue.
```
while True:
    try:
        print("Processing before polling...")
        server.poll()
        print("Processing after polling...")
    except OSError:
       continue
```
The code will stop at the `server.poll()` method waiting for a request to come in...
```
Listening on http://192.168.1.208:80
Processing before polling...
```
Once a request comes in the other lines will process, then it'll stop at `server.poll()` again...
```
Processing after polling...
Processing before polling...
```

To fix this the socket needs to set blocking to False or it's timeout to 0 in `HTTPServer.Start()`...
```
        self._sock.bind((host, port))
        self._sock.listen(5)
        self._sock.setblocking(False) #do not block
```

There is one caveat to this.  When the accept() method doesn't find any incoming connections and blocking is turned off, it raises an exception `OSError [Errno 11] EAGAIN`.  Therefore, `server.poll()` must be the last thing in the while loop otherwise, the exception will skip any logic after the exception is raised.  Which means any logic after `server.poll()`  will only run once a request has been received and processed successfully...
```
while True:
    try:
        print("Processing before polling...")
        server.poll()
        print("Processing after polling...") #This will never be called if exception is raised.
    except OSError:
       continue
```
Results in this output...
```
Listening on http://192.168.1.208:80
Processing before polling...
Processing before polling...
Processing before polling...
Processing before polling...
Received 485 bytes
Processing after polling...
Processing before polling...
Processing before polling...
...  and so on
```

A possible work around for this would be to handle the specific exception `OSError: [Errno 11] EAGAIN` in the `server.poll()` method.  But I'm not sure if catching and handling exceptions in library code is an acceptable practice.

### PR DESCRIPTION
**Pull request by karlfl**
Set socket blocking to false in HTTPServer.poll().  This will allow calling code to continue to process even when there are no web requests coming in.  Without this it would block at the self._sock.accept() waiting for the next request.

Handle EAGAIN & ECONNRESET errors in HTTPServer.poll()  These errors indicate there are no requests coming in (EAGAIN) or the peer has reset the connection (ECONNRESET).  The calling code shouldn't have to handle these conditions as they are part of normal web interactions and can be safely handled in HTTPServer.

Fix example file name to follow standard naming of "modulename_simple.py"

This PR closes #9

### CODE REVIEW

**Comment by karlfl** [comment]
Not sure how to link an issue, but this PR will fix issue #9 - socket.accept() is blocking by default.

**Comment by dhalbert** [comment]
(You can link an issue by clicking the gear icon to the right of "Development" in the right sidebar, or by mentioning the issue in a specific way in the original post: 
https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)

File: adafruit_httpserver.py [hunk 1]
```diff
@@ -327,26 +327,37 @@ def start(self, host: str, port: int = 80, root: str = "") -> None:
...
+                    response = HTTPResponse(status=HTTPStatus.INTERNAL_SERVER_ERROR)
+
+                response.send(conn)
+        except OSError as ex:
+            # handle EAGAIN and ECONNRESET
+            if ex.errno == EAGAIN:
+                # there is no data available right now, try again later".
+                return
+            if ex.errno == ECONNRESET:
+                # connection reset by peer, try again later".
```
Comment (dhalbert) [inline adafruit_httpserver.py:361]: Did you mean to put "later" in quotes? There's only one quotation mark?

**Review by dhalbert — CHANGES_REQUESTED** [review]
Looks good. Just a comment question.

File: adafruit_httpserver.py (hunk 1 above)
Comment (karlfl) [reply adafruit_httpserver.py:361]: Not sure how that " got in there.  Mabe a cut/paste error or something.  I'll fix it.

**Review by dhalbert — APPROVED** [review]
Thanks!

**Pull request merged**